File: app/src/main/java/com/tablehockey/game/network/NetworkSession.kt
package com.tablehockey.game.network

/**
 * Holds the live host / guest link set up in the lobby so GameActivity can
 * pick it up. Sockets can't travel through an Intent, so this process-wide
 * singleton is the hand-off point between the two screens. Works for both
 * the WiFi and the Bluetooth transports.
 */
object NetworkSession {
    var host: HostLink? = null
    var guest: GuestLink? = null

    fun clear() {
        host?.stop()
        guest?.disconnect()
        host = null
        guest = null
    }
}

File: ios/PowerPlayHockeyCore/Sources/PowerPlayHockeyCore/Network/NetworkSession.swift
import Foundation

/// Holds the live host / guest link set up in the multiplayer lobby so GameView can pick it up.
/// Works across both local Bonjour/WiFi and Cloudflare WebSocket internet relay.
public final class NetworkSession {
    public static let shared = NetworkSession()

    public var host: HostLink?
    public var guest: GuestLink?

    private init() {}

    public func clear() {
        host?.stop()
        guest?.disconnect()
        host = null
        guest = null
    }
}
